salesagility/SuiteCRM
# ListView: Fix selection when switch from "select all" to "select page"

## Description

When selecting all list entries and switching between pages and then switching the selection
back to "select page" the code tried to keep the current selection, but this doesn't really work
because the saved uids don't include the whole list, so this results in a random selection depending
on the pages visited between the two selection actions.

Instead keep it simple and when going from "all" to "page": clear the selection state and select the
current page.

## Motivation and Context

Selecting the page shouldn't select/deselect random other pages.

## How To Test This

1) "Select all"
2) Next page
3) Previous page
4) "Select page"
5) Only this page should be selected (previously the first two pages were selected, and none after that)

## Types of changes

### Final checklist
